sql/sql_alter.cc

```diff
@@ -346,6 +347,43 @@ bool Sql_cmd_alter_table::execute(THD *thd) {
   if (check_grant(thd, priv_needed, first_table, false, UINT_MAX, false))
     return true; /* purecov: inspected */
 
+#ifdef WITH_WSREP
+  /* Check if foreign keys are accessible.
+  1. Transaction is replicated first, then is done locally.
+  2. Replicated node applies write sets in context
+  of root user.
+  Above two conditions may cause that even if we have no access to FKs,
+  transactions will replicate with success, but fail locally.
+
+  Here we check only for FK access, not if the engine supports FKs.
+  That's enough, because right now we need only to know if transaction
+  should be rolled back because of lack of access and not replicated,
+  or we can replicate it and let replicated node do the rest of the job. */
+  if (alter_info.flags & Alter_info::ADD_FOREIGN_KEY) {
+    if (check_fk_parent_table_access(thd, &create_info, &alter_info, false)) {
+      return true;
+    }
+  }
+
+  TABLE *find_temporary_table(THD * thd, const TABLE_LIST *tl);
```

Comment:
Not a strong opinion. From the point of readability, can we change it to 
```suggestion
 extern TABLE *find_temporary_table(THD * thd, const TABLE_LIST *tl);
```
just like https://github.com/kamil-holubicki/percona-xtradb-cluster/blob/PXC-3449/sql/sql_alter.cc#L601

Reply:
Out of the scope of the current work, but I agree. Changed.